### ISSUE TITLE
Does anybody has experience migrating a production app from this library to passport?

### ISSUE THREAD

**Issue by kalemdzievski** [tymondesigns/jwt-auth#2145]
It is incredible how terribly maintained this library is, yet so used.
The last version release is almost a year ago. 500+ issues opened...

Does anybody have experience migrating a production app to some other library like a passport?
It would be great if we could share experiences.
Thanks!

**Comment by Messhias** [comment]
@kalemdzievski unfortunately the migration for us that we already started using this one is very bad since there's another kind of configuration of passport. 

And for my option the passport it's not soo quite good to handle multiples middleware, for my example, at our company we have an external API for our customers that can be integrated into our system and this library worked very well.

I tried to enter in contact with @tymondesigns to help him maintain the library (I also contributed for it with #983 opening the MR #2139 to avoid this annoying message (our logs in production went crazy because of this).

And I still waiting for his response.

**Comment by kalemdzievski** [comment]
It would definitely be a pickle to migrate to some other library midflight on the production environment.
There would be a lot of code to handle this. Not sure if it is possible in the first place. I would share any research whenever I get a chance to do it.

**Comment by Messhias** [comment]
> It would definitely be a pickle to migrate to some other library midflight on the production environment.
> There would be a lot of code to handle this. Not sure if it is possible in the first place. I would share any research whenever I get a chance to do it.

Everything depends on you in that case, because if this library continues going in that way (like be more than 2 - 3 years) without support the newest PHP will have a lot of conflicts with some functions, methods, and deprecated methods.

So if you are worried about that, might be it's a good idea to migrate instead of wait until the problems to come to your door when you'll have more a huge load of code to deal with it.

**Comment by ajcastro** [comment]
@Messhias how about creating a new fork of this repository where the maintainer will be you or other persons? is that possible?

**Comment by Messhias** [comment]
> @Messhias how about creating a new fork of this repository where the maintainer will be you or other persons? is that possible?

Definitely is possible. But we need to change the README.md to give the credits of the original for @tymondesigns even he abandoned the original ones. 

I just will need help to create the package in the package, etc to do the right appointments for the composer. 

I never published a library before.

**Comment by ajcastro** [comment]
Hi @Messhias I have experience in creating packages.  I am willing to help. I have just questions like,
What are the steps we need to do? What will be the new package/repository name? How do we deal with the existing opened PRs and issues?

**Comment by Messhias** [comment]
> Hi @Messhias I have experience in creating packages. I am willing to help. I have just questions like,
> What are the steps we need to do? What will be the new package/repository name? How do we deal with the existing opened PRs and issues?

Hi, let's see:

- The first step now is to close the oldest PR (merging it or just close it since could being too old).
- The name to be honest for me I really don't. care hahahahhahaha
- The opened PR's and issues we'll need to solve, there's some but if you see the package it's not like an Unreal Engine which appears PR(s) issue(s) almost every hour, so I believe while we have more maintainers more quickly we'll solve the issues.

**Comment by ajcastro** [comment]
@Messhias , that's great thanks!

For the package name, I think it is important. I remember what happened when the first maintainer of faker-php steps down. The open-source community created a GitHub team which is FakerPHP which houses all the repositories including the docs. I am wondering if we are going to do the same steps. 
If that's the case the new repo name might be `JWT-Auth-PHP/jwt-auth`? 😅😄

**Comment by Messhias** [comment]
> @Messhias , that's great thanks!
> 
> For the package name, I think it is important. I remember what happened when the first maintainer of faker-php steps down. The open-source community created a GitHub team which is FakerPHP which houses all the repositories including the docs. I am wondering if we are going to do the same steps.
> If that's the case the new repo name might be `JWT-Auth-PHP/jwt-auth`? 😅😄

Also, there's another issue to we attempt too about the package, it's about the documentation, now it's maintained by @tymondesigns and he holds everything at his hand, so probably it'll be necessary we address new documentation, for my, could be in the wiki in GitHub it's good enough and easier to maintain, I don't know. This is need be analyzed too.

**Comment by ajcastro** [comment]
@Messhias yeah I think the wiki is enough

**Comment by Messhias** [comment]
> @Messhias yeah I think the wiki is enough

Ok, if you want to, you can create and do all the job, change the namespacing, etc, and once it's everything's set up we can start merging the old merge requests, we have a merge request to review since 2018 and a total of 36... In other words, there's a lot of work to do.

**Comment by ajcastro** [comment]
@Messhias I think I am not a better fit to own the organization for `JWTAuthPHP/jwt-auth`. Maybe we should select someone who knows more about the internals of the package. I guess, between you and me, you know better so you should be the one to hold the organization/package. What do you think of my suggestion?

**Comment by Messhias** [comment]
> @Messhias I think I am not a better fit to own the organization for `JWTAuthPHP/jwt-auth`. Maybe we should select someone who knows more about the internals of the package. I guess, between you and me, you know better so you should be the one to hold the organization/package. What do you think of my suggestion?

@ajcastro I believe overall it's a good idea, but since it's a public organization, let's give a good name, here in Portugal I have a company called Algoritmo Imprevisível Unipessoal LTDA, I believe should be something like and in that way we could fork more old / abandoned packages into it and help those open source packages.

**Comment by eschricker** [comment]
Is there any progress in bringing the package into a new organization?

**Comment by Messhias** [comment]
> Is there any progress in bringing the package into a new organization?

I'm waiting for the @ajcastro answer since so far only shows interested to keep maintaining the package. Did you interested to help us to fork this package?

**Comment by ajcastro** [comment]
Hi @Messhias sorry I was kind of busy this past few days on my work. I think you can call the shot. I will just check what I can help.

**Comment by Messhias** [comment]
> Hi @Messhias sorry I was kind of busy this past few days on my work. I think you can call the shot. I will just check what I can help with.

I believe it's not a smart idea for to me start and keep maintaining this alone otherwise it'll be not so different from this actual package, I could be busy too for few days, we need to start in the right way with 1 - 3 maintainers to keep the package growing.

**Comment by eschricker** [comment]
> I'm waiting for the @ajcastro answer since so far only shows interested to keep maintaining the package. Did you interested to help us to fork this package?

I'm not experienced in publishing packages. Currently I'm using the package in a lumen project. But I can try to help :)

**Comment by Messhias** [comment]
> > I'm waiting for the @ajcastro answer since so far only shows interested to keep maintaining the package. Did you interested to help us to fork this package?
> 
> I'm not experienced in publishing packages. Currently, I'm using the package in a lumen project. But I can try to help :)

I'm not also, but it seems there are at least 3 people interested, but you have experience in PHP and in contributing with open source projects? If yes, I think at least we'll have 3 maintainers, what's sounds like a good deal for the start.

**Comment by eschricker** [comment]
Yes ,I have experience in using PHP. My experience in contributing to open source projects is stil in the beginning :laughing:

**Comment by Messhias** [comment]
> Yes ,I have experience in using PHP. My experience in contributing to open source projects is stil in the beginning 😆

Nice, do you want to create the organization and fork the project to work with it?

**Comment by eschricker** [comment]
I can do this. Do we have a new name for the organization?
You mentioned that it should be more general to be able to host multiple abandoned or old php projects. Do you know a good name?

**Comment by Messhias** [comment]
> I can do this. Do we have a new name for the organization?
> You mentioned that it should be more general to be able to host multiple abandoned or old PHP projects. Do you know a good name?

I don't have suggestion at time, but we could call PHP Open Source Saver 😂👌🏻

Or something like that.

**Comment by eschricker** [comment]
I think the name is great. I have created the Github Organization and added both of you to it. The jwt-auth project is already imported

**Comment by SherinBloemendaal** [comment]
Nice work guys! We are using this project in combination with Laravel Passport. We are missing some core features and we would like to contibute also. Maybe its also a good idea to create a Slack channel for faster communication? 

I also think its good to speed up the process, maybe we can have a version live in 2 weeks?

**Comment by eschricker** [comment]
@SherinBloemendaal I can add you to the organization. Currently I just have transferred the git, but I haven't started to migrate the pull requests yet. 
I have enabled the discussions for the repository. I thought it would be better to keep the communication there. If a slack channel is preferred, thats fine with me.

**Comment by specialtactics** [comment]
I was going to fork this project in my org, but if someone else has already done it and is committed to maintaining and developing it, even better.

Can you guys please share the repo? @eschricker

**Comment by eschricker** [comment]
The repo is https://github.com/PHP-Open-Source-Saver/jwt-auth
@specialtactics if you are want to contribute, I can add you to the organization as well. Till now just the repo has been forked.

**Comment by SherinBloemendaal** [comment]
> @SherinBloemendaal I can add you to the organization. Currently I just have transferred the git, but I haven't started to migrate the pull requests yet.
> I have enabled the discussions for the repository. I thought it would be better to keep the communication there. If a slack channel is preferred, thats fine with me.

Yes that would be nice!

**Comment by eschricker** [comment]
> > @SherinBloemendaal I can add you to the organization. Currently I just have transferred the git, but I haven't started to migrate the pull requests yet.
> > I have enabled the discussions for the repository. I thought it would be better to keep the communication there. If a slack channel is preferred, thats fine with me.
> 
> Yes that would be nice!

I invited you :)

**Comment by mfn** [comment]
The organization name is golden 👍

**Comment by specialtactics** [comment]
Hey @eschricker that would be lovely thanks, we use this package quite a lot for projects in our company, and I also use it personally, so I'm quite invested in making sure it's well supported, and would be happy to help out.

**Comment by Messhias** [comment]
> Nice work guys! We are using this project in combination with Laravel Passport. We are missing some core features and we would like to contibute also. Maybe its also a good idea to create a Slack channel for faster communication?
> 
> I also think its good to speed up the process, maybe we can have a version live in 2 weeks?

good idea.

**Comment by eschricker** [comment]
@specialtactics I invited you. @Messhias and @ajcastro you should also have an open invitation.

**Comment by Messhias** [comment]
> @specialtactics I invited you. @Messhias and @ajcastro you should also have an open invitation.

Hi, sorry but it didn't appear any invitation yet for me, where can I found on GitHub dashboard?

**Comment by eschricker** [comment]
@Messhias I don't know where the invitation is shown. I have never been added to a github organization :D 

Probably the invitation is shown when you visit https://github.com/PHP-Open-Source-Saver ?

**Comment by Messhias** [comment]
> @Messhias I don't know where the invitation is shown. I have never been added to a github organization :D
> 
> Probably the invitation is shown when you visit https://github.com/PHP-Open-Source-Saver ?

yes, exactly the invitation shows there, but unfortunately, it's already expired when I tried to enter on you, can you resend the invitation for me, please? 

Thank you,

**Comment by eschricker** [comment]
I have resend the invitation

**Comment by specialtactics** [comment]
hey @eschricker sorry I was away due to personal issues, do you mind resending it to me? Thanks.

**Comment by eschricker** [comment]
@specialtactics I have resend the invitation :)

**Comment by specialtactics** [comment]
@eschricker thanks buddy!
